try {
    Remove-Item -Path 'c:\IDoNotExist.txt' -ErrorAction Stop
} catch  [System.Management.Automation.ItemNotFoundException] {
    ""[System.Management.Automation.ItemNotFoundException]""
    throw #though we caught the exception we're now rethrowing it.  This is useful in scenarios such as where we want to log that an exception's occurred, but still allow the exception to bubble up to a higher layer
} catch [System.Exception] {
    ""[System.Exception]""
} finally {
    ""Always do this after a try/catch, regardless of whether there was an error, we caught the error, or we rethrew the error""
}
""If any uncaught error exists (i.e. including rethrown errors from the catch block) we won't reach this... If there were no errors or all errors were handled and not rethrown, we will""
